managedtenants/bundles/bundle.py

```diff
@@ -69,29 +89,44 @@ def _validate_operator_sdk(self):
                 f" {_OPERATOR_SDK.version}: {e.stdout.decode()}."
             )
 
-    def _parse_metadata_annotations(self):
-        annotations_file = self.path / "metadata" / "annotations.yaml"
-        with open(annotations_file, "r", encoding="utf-8") as f:
-            try:
-                data = yaml.load(f, Loader=yaml.CSafeLoader)
-
-            except FileNotFoundError as e:
-                raise BundleError(f"{self}: {annotations_file} not found: {e}")
+    def _validate_csv(self):
+        # 1. enforce the `olm.skipRange` annotation
```

Comment:
want to clarify. either skiprange or replaces should be present?, and this is enforcing skiprange to be present?

Reply:
It's a pattern where `replaces` can never be specified, and the `olm.skipRange` annotation has to be present. This is to make sure we define a proper upgrade graph from OLM's perspective. https://olm.operatorframework.io/docs/concepts/olm-architecture/operator-catalog/creating-an-update-graph/

Reply:
This pattern is used by OCS currently (see their CSV's in managed-tenants-bundles).